app/controllers/trains_controller.rb

```diff
@@ -7,14 +7,23 @@ def index
     @trains = TrainApi.new.index
   end
 
+  def create
+    response = TrainApi.new.create(params[:id], train_params)
+
+    if response.success?
+      redirect_to trains_path, notice: response["message"]
+    else
+      flash["errors"] = response["errors"]
+      @train = params[:train]
+      render "new"
```

Comment:
where is the new action?

Reply:
> where is the new action?

the api doesnt have a `new` route tho. do i create a route for that in the api?

Reply:
strangely the new path all works without the new action

Reply:
this isnt the api dude.
this is an app thats talking to the api.
this needs its own routes for displaying views, i put a note on why this is magically working on some other parts.
But you gotta try to understand the difference between the api and this app